internal/clients/packer.go

```diff
@@ -48,6 +48,96 @@ func GetIterationFromID(ctx context.Context, client *Client, loc *sharedmodels.H
 	return it.Payload.Iteration, nil
 }
 
+// ChannelIterationAssignment
+type ChannelIterationAssignment struct {
+	IterationID                 string
+	IterationFingerprint        string
+	IterationIncrementalVersion int
+}
+
+// CreateBucketChannel creates a channel on the named bucket.
+func CreateBucketChannel(ctx context.Context, client *Client, loc *sharedmodels.HashicorpCloudLocationLocation, bucketSlug, channelSlug string) (*packermodels.HashicorpCloudPackerChannel, error) {
+	params := packer_service.NewPackerServiceCreateChannelParamsWithContext(ctx)
+	params.LocationOrganizationID = loc.OrganizationID
+	params.LocationProjectID = loc.ProjectID
+	params.BucketSlug = bucketSlug
+	params.Body.Slug = channelSlug
+
+	channel, err := client.Packer.PackerServiceCreateChannel(params, nil)
+	if err != nil {
+		err := err.(*packer_service.PackerServiceCreateChannelDefault)
+		return nil, errors.New(err.Payload.Message)
+	}
+
+	return channel.GetPayload().Channel, nil
+}
+
+// UpdateBucketChannel updates the named channel.
+func UpdateBucketChannel(ctx context.Context, client *Client, loc *sharedmodels.HashicorpCloudLocationLocation, bucketSlug string, channelSlug string,
+	assignment *ChannelIterationAssignment) (*packermodels.HashicorpCloudPackerChannel, error) {
+	params := packer_service.NewPackerServiceUpdateChannelParamsWithContext(ctx)
+	params.LocationOrganizationID = loc.OrganizationID
+	params.LocationProjectID = loc.ProjectID
+	params.BucketSlug = bucketSlug
+	params.Slug = channelSlug
+
+	if assignment != nil {
+		switch {
+		case assignment.IterationID != "":
+			params.Body.IterationID = assignment.IterationID
+		case assignment.IterationFingerprint != "":
+			params.Body.Fingerprint = assignment.IterationFingerprint
+		case assignment.IterationIncrementalVersion > 0:
+			params.Body.IncrementalVersion = int32(assignment.IterationIncrementalVersion)
+		}
+	}
+
+	channel, err := client.Packer.PackerServiceUpdateChannel(params, nil)
+	if err != nil {
+		err := err.(*packer_service.PackerServiceUpdateChannelDefault)
+		return nil, errors.New(err.Payload.Message)
+	}
+
+	return channel.GetPayload().Channel, nil
+}
+
+// DeleteBucketChannel deletes a channel from the named bucket.
+func DeleteBucketChannel(ctx context.Context, client *Client, loc *sharedmodels.HashicorpCloudLocationLocation, bucketSlug, channelSlug string) (*packermodels.HashicorpCloudPackerChannel, error) {
+	params := packer_service.NewPackerServiceDeleteChannelParamsWithContext(ctx)
+	params.LocationOrganizationID = loc.OrganizationID
+	params.LocationProjectID = loc.ProjectID
+	params.BucketSlug = bucketSlug
+	params.Slug = channelSlug
+
+	req, err := client.Packer.PackerServiceDeleteChannel(params, nil)
+	if err != nil {
+		err := err.(*packer_service.PackerServiceDeleteChannelDefault)
+		return nil, errors.New(err.Payload.Message)
+	}
+
+	if !req.IsSuccess() {
```

Comment:
Why the other requests don't have this check?

Reply:
Other requests return a channel which is nil checked by the caller before trying to continue. Since the call to Delete does not actually return a channel IsSuccess is used to verify that the delete request was successful.

Reply:
Do you think it is valuable to call IsSuccess after each Channel create/update request?

Reply:
Got it! No need to change the others, the reason makes sense to me. I didn't know about that method